src/Analyzers/Core/Analyzers/UseObjectInitializer/UseNamedMemberInitializerAnalyzer.cs

```diff
@@ -77,11 +87,13 @@ protected override void AddMatches(ArrayBuilder<Match<TExpressionSyntax, TStatem
                 }
             }
 
-            foreach (var child in containingBlock.ChildNodesAndTokens())
+            foreach (var child in containingBlockOrCompilationUnit.ChildNodes())
```

Comment:
so ChildNodes is more expensive thank ChildNodesAndTokens.  We use the former to prevent allocations and interface dispatch .  tnx :)